docs/process/style_guide.md

```diff
@@ -608,13 +608,14 @@ Main issue: [#113](https://github.com/unicode-org/icu4x/issues/113)
 
 When structs with public string fields contain strings, use the following type conventions:
 
-- If lifetime parameters are allowed, use `&'a str`.
-- If lifetime parameters are not allowed, use one of:
-  - [TinyStr](https://github.com/zbraniecki/tinystr) if the string is ASCII-only.
-  - [SmallString](https://crates.io/crates/smallstr) for shorter strings, with a stack size ∈ {8, 12, 16, 20} that fits at least 99% of cases.
-  - `Cow<'static, str>`  for longer strings.
+- `&'s str` if the struct does not need to own the string.
+- `Cow<'s, str>`  if the string could be borrowed or owned.
```

Comment:
We should probably weigh the lifetime complexity against other things here, @zbraniecki can attest that zero-copy-everything can lead to some gnarly constraints, and we might want to mention "if you think it won't make lifetimes too complex" here. In many cases it'll be fine, but if central types started doing this we'd have problems.